hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/test/java/org/hypertrace/traceenricher/enrichment/enrichers/ResourceAttributeEnricherTest.java

```diff
@@ -0,0 +1,177 @@
+package org.hypertrace.traceenricher.enrichment.enrichers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.hypertrace.core.datamodel.*;
+import org.hypertrace.traceenricher.enrichment.clients.ClientRegistry;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ResourceAttributeEnricherTest extends AbstractAttributeEnricherTest {
+
+  private final ResourceAttributeEnricher resourceAttributeEnricher =
+      new ResourceAttributeEnricher();
+
+  @BeforeAll
+  public void setup() {
+    String configFilePath =
+        Thread.currentThread().getContextClassLoader().getResource("enricher.conf").getPath();
+    if (configFilePath == null) {
+      throw new RuntimeException("Cannot find enricher config file enricher.conf in the classpath");
+    }
+
+    Config fileConfig = ConfigFactory.parseFile(new File(configFilePath));
+    Config configs = ConfigFactory.load(fileConfig);
+    if (!configs.hasPath("enricher.ResourceAttributeEnricher")) {
+      throw new RuntimeException(
+          "Cannot find enricher config for ResourceAttributeEnricher in " + configs);
+    }
+    resourceAttributeEnricher.init(
+        configs.getConfig("enricher.ResourceAttributeEnricher"), mock(ClientRegistry.class));
+  }
+
+  @Test
+  public void noAttributes() {
+    Event event = mock(Event.class);
+    when(event.getAttributes()).thenReturn(null);
+    resourceAttributeEnricher.enrichEvent(null, event);
+
+    when(event.getAttributes()).thenReturn(Attributes.newBuilder().build());
+    when(event.getAttributes().getAttributeMap()).thenReturn(null);
+    resourceAttributeEnricher.enrichEvent(null, event);
+  }
+
+  @Test
+  public void noResourceInTrace() {
+    // This trace has no resource attributes.
+    StructuredTrace trace = getBigTrace();
+    for (Event event : trace.getEventList()) {
+      int attributeMapSize = 0;
+      if (event.getAttributes() != null && event.getAttributes().getAttributeMap() != null) {
+        attributeMapSize = event.getAttributes().getAttributeMap().size();
+      }
+      resourceAttributeEnricher.enrichEvent(trace, event);
+      if (event.getAttributes() != null && event.getAttributes().getAttributeMap() != null) {
```

Comment:
Do you really need these checks if you're constructing the trace yourself?

Reply:
Yes. These checks were not before and some of the events of custom trace did not have these attributes which makes sense to test on different edge cases.